### PR TITLE
Refactored test_resources.py into test_modelresources package

### PR DESCRIPTION
**Pull request by bvedang**
**Problem**

What problem have you solved?

The `test_resources.py` module exceeded 1.8k lines, becoming difficult to navigate and manage over time as more resource tests were added.

**Solution**

How did you solve the problem?

- Added `test_modelresources` package

- Split `ModelResource` tests into multiple classes.

**Acceptance Criteria**

Have you written tests? Have you included screenshots of your changes if applicable?
Did you document your changes? 

Tests executed against test_resources package to confirm no change in overall coverage or results.
No UI or documentation changes were required for this strictly organizational refactor.

### CODE REVIEW

File: tests/core/tests/test_resources/test_modelresource/test_data_deletion_dry_run.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+import tablib
+from core.models import Book
+from core.tests.resources import BookResource
+from core.tests.utils import ignore_widget_deprecation_warning
+from django.test import TestCase
+
+from import_export import fields, results, widgets
+
+
+class DataDeletionDryRunTests(TestCase):
```
Comment (matthewhegarty) [inline tests/core/tests/test_resources/test_modelresource/test_data_deletion_dry_run.py:12]: Please can we keep the convention of using "Test" not "Tests".  For example, I prefer that this name is `DataDeletionDryRunTest`.  Please can you modify other test cases to end with `Test` and not `Tests`

**Review by matthewhegarty — CHANGES_REQUESTED** [review]
Thanks for doing this.  It is much better to have smaller modules like you have added.  I have made a couple of comments which I think would improve the PR.

File: tests/core/tests/test_resources/test_modelresource/test_data_deletion_dry_run.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    def setUp(self):
+        self.resource = BookResource()
+        self.book = Book.objects.create(name="Some book")
+        self.dataset = tablib.Dataset(headers=["id", "name", "author_email", "price"])
+        row = [self.book.pk, "Some book", "test@example.com", "10.25"]
+        self.dataset.append(row)
+
+    @ignore_widget_deprecation_warning
+    def test_import_data_delete(self):
```
[the remark this answers is not in the document]
Comment (bvedang) [reply tests/core/tests/test_resources/test_modelresource/test_data_deletion_dry_run.py]: Certainly, your suggestion to rename the module to` test_data_deletion.py `makes sense, especially if not all tests within the `test_data_deletion_dry_run.py` module use the dry run approach. It's indeed important to have module names that accurately reflect the tests they contain. I have tried to name files based on the specific tests they are intended to perform, but I appreciate your input on this. I will review the module names and make sure they align more closely with the tests they include. Thank you for bringing this to my attention.

[the remark this answers is not in the document]
Comment (matthewhegarty) [reply tests/core/tests/test_resources/test_modelresource/test_data_deletion_dry_run.py]: Thanks - please let me know when you would like me to review

**Comment by bvedang** [comment]
Hi @matthewhegarty 

I have removed the `s` from the class name in updated refactored test classes as per the instruction and also rename the file s

**Comment by bvedang** [comment]
Sure, thank you for resolving the conflicts

**Comment by matthewhegarty** [comment]
I'm happy to merge this as it is.  I would like to do so soon because it is blocking other work.  I will hold off if you want to make more changes.  Please can you let me know.

**Comment by bvedang** [comment]
Hi @matthewhegarty , I was trying to figure out what do i need to take care of, i am confused. Can you help me with it
Thank you

**Comment by matthewhegarty** [comment]
Hi - it's all good - I will merge.  Thanks for all your help.

**Pull request merged**